src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -1078,9 +1078,9 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
     /// works because C# is pass by reference for referenced class types.
     /// </summary>
     /// <param name="invalidColumns">List in which to aggregate the invalid fields.</param>
-    /// <param name="fields">List of the fields to check for existence in backing DB.</param>
+    /// <param name="fields">List of the backing fields to check for existence in backing DB.</param>
     /// <param name="entityName">The name of the entity that we check for backing columns.</param>
-    /// <param name="sqlMetadataProvider">The sqlMetadataProvider used to lookup if the fields are valid columns in DB.</param>
+    /// <param name="sqlMetadataProvider">The sqlMetadataProvider used to lookup if the backing fields are valid columns in DB.</param>
```

Comment:
The keyset of that dictionary includes every backing column of every entity, if that backing columns has no alias in the config then it will be mapped to itself (eg. id --> id). This is the best way to do the validation now, when we add the linking object validation (since linking objects may not be entities) I will explore adding a new data structure to hold these values, or modifying one we already use.

Reply:
> The keyset of that dictionary includes every backing column of every entity

Understood, just confirming that the keyset is sourced from SourceDefinitions.

Reply:
Yup, this is the actual function that handles it if you are curious

![image](https://github.com/Azure/data-api-builder/assets/93220300/176415f7-71a6-42e7-8b35-22c17fdf8803)